src/platform/Linux/bluez/BluezAdvertisement.cpp

```diff
@@ -170,6 +143,53 @@ CHIP_ERROR BluezAdvertisement::Init(const BluezEndpoint & aEndpoint, ChipAdvType
     return err;
 }
 
+CHIP_ERROR BluezAdvertisement::SetIntervals(AdvertisingIntervals aAdvIntervals)
+{
+    VerifyOrReturnError(mpAdv != nullptr, CHIP_ERROR_UNINITIALIZED);
+    // If the advertisement is already running, BlueZ will update the intervals
+    // automatically. There is no need to stop and restart the advertisement.
+    bluez_leadvertisement1_set_min_interval(mpAdv, aAdvIntervals.first * 0.625);
+    bluez_leadvertisement1_set_max_interval(mpAdv, aAdvIntervals.second * 0.625);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR BluezAdvertisement::SetupServiceData(ServiceDataFlags aFlags)
+{
+    VerifyOrReturnError(mpAdv != nullptr, CHIP_ERROR_UNINITIALIZED);
+
+    Ble::ChipBLEDeviceIdentificationInfo deviceInfo;
+    ReturnErrorOnFailure(ConfigurationMgr().GetBLEDeviceIdentificationInfo(deviceInfo));
+
+#if CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING
+    deviceInfo.SetAdditionalDataFlag(true);
+#endif
+
+#if CHIP_DEVICE_CONFIG_BLE_EXT_ADVERTISING
+    if (aFlags & kServiceDataExtendedAnnouncement)
```

Comment:
I suppose this is also kind of a spec question, but is this flag meant to be set in the service data payload even during the initial announcement? If so, keying the VID/PID elision off of this flag is incorrect since the VID and PID then won't appear during the initial announcement duration.

Reply:
To be honest I don't know. The spec is not very clear about that (5.4.2.3.1. Extended Announcement). But I assume that the extended announcement can be "enabled" only after the normal announcement period. I've based that particular logic on the original PR: https://github.com/project-chip/connectedhomeip/pull/31532

Reply:
Test plan uses that flag (I think? - also a bit unclear) during the initial duration to determine what the device is going to do after 15 mins. So with this change, you're going to fail the cert test. (https://github.com/CHIP-Specifications/chip-test-plans/pull/3869/files) From a client POV, I think it makes sense to have that flag on the whole time since then the client knows if this thing is going to disappear after 15 mins.

@ndyck14 - can you clarify the intent here? This might also need a spec issue.

Reply:
the purpose of the flag is to be used _during_ the extended period, not during the initial 15 minutes. This is to allow any prospective listeners to make intelligent decisions on whether a newly detected device is likely something that a user has recently plugged in with the immediate intent to setup (e.g. proactive notifications to users)

Reply:
oh. OK, yeah, I totally did not get that from the spec at all. And that means the cert test is wrong.

Reply:
I mean...kind of. Might just be my misreading.
```
When using Extended Announcement, the device SHALL set the Extended Announcement Flag in the Matter Service Data in the BLE Advertisement (see Table 50, “Matter BLE Service Data payload format”).
```
I read that as when the device SUPPORTS extended announcement, but it probably means "during the extended announcement period"